src/core/algorithms/statistics/data_stats.h

```diff
@@ -21,9 +21,13 @@ class DataStats : public Algorithm {
 
     void ResetState() final;
 
-    // Returns number of elements in the column satisfying the predicate
-    template <class Pred>
-    size_t CountIf(Pred pred, size_t index) const;
+    // Returns number of elements satisfying the predicate
+    template <class Pred, class Data>
+    size_t CountIf(Pred pred, const Data& data) const;
+
+    // Returns vector with indices satisfying the predicate
+    template <class Pred, class Data>
+    std::vector<size_t> GetIndices(Pred pred, const Data& data) const;
```

Comment:
I'd rename this function to something like `FilterIndices`, because what this function essentially does is implements [`ranges::views::filter`](https://en.cppreference.com/w/cpp/ranges/filter_view), but with indices returned.